packages/flutter_tools/lib/src/resident_devtools_handler.dart

```diff
@@ -91,6 +92,30 @@ class FlutterResidentDevtoolsHandler implements ResidentDevtoolsHandler {
     final List<FlutterDevice?> devicesWithExtension = await _devicesWithExtensions(flutterDevices);
     await _maybeCallDevToolsUriServiceExtension(devicesWithExtension);
     await _callConnectedVmServiceUriExtension(devicesWithExtension);
+
```

Comment:
Wondering the reason for putting this in `serveAndAnnounceDevTools()`. Can this code be pulled out into its own method?

Reply:
Only because this code is only called in this method, whereas the functions above are called in a couple of places. I can pull this into a helper method if you'd prefer.

Reply:
I guess what I'm trying to learn is whether serving the observatory has a real dependence on the DevTools-specific lines above in this method, or if this code could be moved out and called independently of `serveAndAnnounceDevTools`? If it's the second, then why not do it that way?

Reply:
Ah okay, I see what you're saying now. I've moved the logic into its own method in `ResidentRunner`.